test/test_tagging.py

```diff
@@ -0,0 +1,110 @@
+    AzulUnitTestCase,
+)
+
+
+def setupModule():
+    configure_test_logging()
+
+
+class TestTagging(AzulUnitTestCase):
+
+    @patch('subprocess.run', new_callable=PropertyMock)
+    def test(self, terraform_mock):
+        # We mock Terraform since it's not available on Travis. The mocked
+        # response only includes resources used by this test case, so we don't
+        # need to update it manually.
+        terraform_mock.return_value.stdout = json.dumps({
+            'format_version': '0.1',
+            'provider_schemas': {
+                'aws': {
+                    'resource_schemas': {
+                        'aws_vpc': {
+                            'block': {
+                                'attributes': {
+                                    'tags': {}
+                                }
+                            }
+                        }
+                    }
+                },
+                'gcp': {
+                    'resource_schemas': {
+                        'google_compute_instance': {
+                            'block': {
+                                'attributes': {
+                                    'tags': {}
+                                }
+                            }
+                        }
+                    }
+                }
+            }
+        }).encode()
+
+        tagged_aws_resource = {
+            'resource': {
+                'aws_vpc': {
+                    'name': {}
+                }
+            }
+        }
+        res = populate_tags(tagged_aws_resource)
+        tags = res['resource'][0]['aws_vpc'][0]['name']['tags']
+        self.assertIn('Name', tags)
+        self.assertNotIn('name', tags)
+
+        tagged_gcp_resource = {
+            'resource': {
+                'google_compute_instance': {
+                    'name': {}
+                }
+            }
+        }
+        res = populate_tags(tagged_gcp_resource)
+        tags = res['resource'][0]['google_compute_instance'][0]['name']['tags']
+        self.assertIn('name', tags)
+        self.assertNotIn('Name', tags)
+
+        untaggable_aws_resource = {
+            'resource': {
+                'aws_untaggable_resource': {
+                    'name': {}
+                }
+            }
+        }
+        res = populate_tags(untaggable_aws_resource)
+        self.assertNotIn('tags', res['resource'][0]['aws_untaggable_resource'])
+
+        manually_tagged_resource = {
+            'resource': {
+                'aws_vpc': {
+                    'name': {
+                        'tags': {
+                            'component': 'foo'
+                        }
+                    }
+                }
+            }
+        }
+        res = populate_tags(manually_tagged_resource)
+        component = res['resource'][0]['aws_vpc'][0]['name']['tags']['component']
+        self.assertEqual(component, 'foo')
+
+
+if __name__ == '__main__':
+    import unittest
```

Comment:
Why is this a local import? At this point that module has already been imported. You can verify that by temporarily adding `assert 'unittest' in sys.modules`.